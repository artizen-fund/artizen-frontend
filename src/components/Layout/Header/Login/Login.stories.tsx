import Login from './'

export default {
  title: 'components/Login',
  component: Login,
  argTypes: {},
}

export const LoginComponent = (props: any) => <Login {...props} />
