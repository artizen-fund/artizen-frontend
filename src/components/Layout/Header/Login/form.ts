import { JsonSchema } from '@jsonforms/core'

export const schema: JsonSchema = {
  type: 'object',
  properties: {
    email: {
      type: 'string',
      format: 'email',
    },
  },
  required: ['email'],
}

export interface FormState {
  email?: string
}

export const initialState: FormState = {}

export const uischema = {
  type: 'VerticalLayout',
  elements: [
    {
      type: 'Control',
      scope: '#/properties/email',
      label: 'Email Address',
      options: { placeholder: 'e.g. rene@artizen.fund' },
    },
  ],
}
