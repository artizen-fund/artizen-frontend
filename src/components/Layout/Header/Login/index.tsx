import { useMemo, useState } from 'react'
import Link from 'next/link'
import styled from 'styled-components'
import { Button, Icon, Form, CheckboxControl } from '@components'
import { rgba } from '@lib'
import { palette, typography } from '@theme'
import { schema, uischema, initialState, FormState } from './form'

const Login = () => {
  const LOCALSTORAGE_KEY = 'loginForm'

  const [data, setData] = useState<FormState>(initialState)
  useMemo(() => {
    if (typeof localStorage === 'undefined') {
      return
    }
    const frozenAnswers = localStorage.getItem(LOCALSTORAGE_KEY)
    if (!frozenAnswers) {
      setData(initialState)
      return
    }
    const thawedAnswers = JSON.parse(frozenAnswers)
    setData(thawedAnswers)
  }, [])

  const [submitted, setSubmitted] = useState(false)
  const [error, setError] = useState<string>()
  const [readonly, setReadonly] = useState(false)
  const [acceptedToc, setAcceptedToc] = useState(true)

  const submit = () => alert('derp')

  return (
    <Wrapper>
      <Headline>Let’s get started by setting up your Artizen account</Headline>
      <InfoRow>
        <Icon glyph="info" outline level={1} />
        <SignInDirections>
          Already have an account?
          <br />
          You can still use this form, it’s magic!
        </SignInDirections>
      </InfoRow>
      <Form localStorageKey={LOCALSTORAGE_KEY} {...{ schema, uischema, initialState, data, setData, readonly }}>
        <Button stretch onClick={() => submit()} disabled={!data.email || !acceptedToc}>
          Sign In / Sign Up
        </Button>
      </Form>
      <OrLine />
      <Alternatives>
        <Button outline onClick={() => alert('derp')} stretch>
          Telegram
        </Button>
        <Button outline onClick={() => alert('derp')} stretch>
          Twitter
        </Button>
        <Button outline onClick={() => alert('derp')} stretch>
          Discord
        </Button>
      </Alternatives>
      <TocCheck>
        <CheckboxControl
          data={acceptedToc}
          path="not-used"
          handleChange={() => setAcceptedToc(!acceptedToc)}
          label=""
        />
        <TocMessage>
          I agree to Artizen’s
          <Link href="/toc">
            <a> Terms &amp; Conditions</a>
          </Link>
        </TocMessage>
      </TocCheck>
    </Wrapper>
  )
}

const Wrapper = styled.div`
  background: ${rgba(palette.white)};
  @media (prefers-color-scheme: dark) {
    background: ${rgba(palette.slate)};
  }
`

const Headline = styled.h1`
  ${typography.title.l2};
`

const InfoRow = styled.div`
  display: flex;
  flex-direction: row;
  gap: 10px;
  justify-content: flex-start;
  align-items: flex-start;
  margin: 1em 0 2em 0;
`

const SignInDirections = styled.p`
  ${typography.label.l1};
`

const OrLine = styled.div`
  position: relative;
  margin: 1em 0;

  text-align: center;
  ${typography.label.l1}
  color: ${rgba(palette.barracuda)};
  text-transform: uppercase;

  &:before {
    content: ' ';
    position: absolute;
    z-index: 0;
    left: 0;
    top: 40%;
    width: 100%;
    height: 1px;
    background: ${rgba(palette.stone)};
  }
  &:after {
    position: relative;
    z-index: 1;
    content: ' or ';
    margin-inline: auto;
    padding: 0 10px;
    background: ${rgba(palette.white)};
  }
`

const Alternatives = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  gap: 10px;
`

const TocCheck = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  a {
    text-decoration: underline;
    text-decoration-thickness: 2px;
  }
`

const TocMessage = styled.p`
  ${typography.label.l1}
  color: ${rgba(palette.barracuda)};
`

export default Login
