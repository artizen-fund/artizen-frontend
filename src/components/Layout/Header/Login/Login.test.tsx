import renderer from 'react-test-renderer'
import Login from './'

describe('Login', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
