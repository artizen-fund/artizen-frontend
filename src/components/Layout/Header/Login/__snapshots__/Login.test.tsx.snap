// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="sc-cf1b6002-0 dLyNyX"
>
  <h1
    className="sc-cf1b6002-1 hUHuMR"
  >
    Let’s get started by setting up your Artizen account
  </h1>
  <div
    className="sc-cf1b6002-2 dsHacc"
  >
    <div
      className="sc-8c7bf1ae-2 huxvCL"
    >
      <div
        className="sc-8c7bf1ae-0 bhoRgU"
        color="night"
      >
        <div
          className="sc-9eb58f28-0 kSymdK"
          color="night"
        />
      </div>
    </div>
    <p
      className="sc-cf1b6002-3 bDtHVf"
    >
      Already have an account?
      <br />
      You can still use this form, it’s magic!
    </p>
  </div>
  <div
    className="vertical-layout"
    hidden={false}
  >
    <div
      className="vertical-layout-item"
    >
      <div
        className="sc-2f9d5146-0 jjUILZ"
        id="#/properties/email"
      >
        <div
          className="sc-233fee17-0 kLNnFl"
        >
          <input
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="e.g. rene@artizen.fund"
            required={true}
            type="text"
          />
          <label
            className="sc-ffdb2e6c-0 eexGRl"
          >
            Email Address
             *
          </label>
        </div>
        <div
          className="sc-93ba83c0-0 bAVzHI"
        />
      </div>
    </div>
  </div>
  <button
    className="sc-a6b3add2-2 hsAXIm undefined  disabled"
    disabled={true}
    onClick={[Function]}
  >
    <span>
      Sign In / Sign Up
    </span>
  </button>
  <div
    className="sc-cf1b6002-4 gsXijY"
  />
  <div
    className="sc-cf1b6002-5 fAZZfH"
  >
    <button
      className="sc-a6b3add2-2 jEHJsK undefined "
      onClick={[Function]}
    >
      <span>
        Telegram
      </span>
    </button>
    <button
      className="sc-a6b3add2-2 jEHJsK undefined "
      onClick={[Function]}
    >
      <span>
        Twitter
      </span>
    </button>
    <button
      className="sc-a6b3add2-2 jEHJsK undefined "
      onClick={[Function]}
    >
      <span>
        Discord
      </span>
    </button>
  </div>
  <div
    className="sc-cf1b6002-6 kBnJzi"
  >
    <label
      className="sc-dd35d48e-4 bqDNzr"
    >
      <div
        className="sc-dd35d48e-0 lmHjtw"
      >
        <input
          checked={true}
          className="sc-dd35d48e-1 clTMzK"
          disabled={false}
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <span
          className="sc-dd35d48e-2 dtceeR"
        />
      </div>
      <span
        className="sc-dd35d48e-3 eAaGSv"
      />
    </label>
    <p
      className="sc-cf1b6002-7 edFTkn"
    >
      I agree to Artizen’s
      <a
        href="/toc"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
         Terms & Conditions
      </a>
    </p>
  </div>
</div>
`;
